omnitool/omnibox/vm.linux/buildcontainer/entry.sh

```
@@ -0,0 +1,345 @@
+#!/bin/bash
+
+# Create User
+USER=${USER:-root}
+HOME=/root
+if [ "$USER" != "root" ]; then
+    echo "* enable custom user: $USER"
+    useradd --create-home --shell /bin/bash --user-group --groups adm,sudo "$USER"
+    echo "$USER ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers
+    if [ -z "$PASSWORD" ]; then
+        echo "  set default password to \"ubuntu\""
+        PASSWORD=ubuntu
+    fi
+    HOME="/home/$USER"
+    echo "$USER:$PASSWORD" | /usr/sbin/chpasswd 2> /dev/null || echo ""
+    cp -r /root/{.config,.gtkrc-2.0,.asoundrc} "$HOME" 2>/dev/null
+    chown -R "$USER:$USER" "$HOME"
+    [ -d "/dev/snd" ] && chgrp -R adm /dev/snd
+fi
+
+# VNC password
+VNC_PASSWORD=${PASSWORD:-ubuntu}
+
+mkdir -p "$HOME/.vnc"
+echo "$VNC_PASSWORD" | vncpasswd -f > "$HOME/.vnc/passwd"
+chmod 600 "$HOME/.vnc/passwd"
+chown -R "$USER:$USER" "$HOME"
+sed -i "s/password = WebUtil.getConfigVar('password');/password = '$VNC_PASSWORD'/" /usr/lib/novnc/app/ui.js
+
+# xstartup
+XSTARTUP_PATH="$HOME/.vnc/xstartup"
+cat << EOF > "$XSTARTUP_PATH"
+#!/bin/sh
+unset DBUS_SESSION_BUS_ADDRESS
+mate-session
+EOF
+chown "$USER:$USER" "$XSTARTUP_PATH"
+chmod 755 "$XSTARTUP_PATH"
+
+# vncserver launch
+VNCRUN_PATH="$HOME/.vnc/vnc_run.sh"
+cat << EOF > "$VNCRUN_PATH"
+#!/bin/sh
+
+# Workaround for issue when image is created with "docker commit".
+# Thanks to @SaadRana17
+# https://github.com/Tiryoh/docker-ros2-desktop-vnc/issues/131#issuecomment-2184156856
+
+if [ -e /tmp/.X1-lock ]; then
+    rm -f /tmp/.X1-lock
+fi
+if [ -e /tmp/.X11-unix/X1 ]; then
+    rm -f /tmp/.X11-unix/X1
+fi
+
+# Create log directory
+mkdir -p "$HOME/.vnc/logs"
+
+if [ $(uname -m) = "aarch64" ]; then
+    LD_PRELOAD=/lib/aarch64-linux-gnu/libgcc_s.so.1 vncserver :1 -fg -geometry 1920x1080 -depth 24"
```

There’s an extra double quote (") at the end of the command, that stray " causes the script to silently fail